group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -709,6 +727,16 @@ public void complete(Throwable exception) {
             }
         }
 
+        @Override
+        public long enqueueTimeMs() {
+            return this.enqueueTimeMs;
+        }
+
+        @Override
+        public void setEnqueueTimeMs(long enqueueTimeMs) {
+            this.enqueueTimeMs = enqueueTimeMs;
+        }
```

Comment:
Instead of setting it like this, would it be possible to just let every event have a created time ms? The event is usually created just before being enqueued so it would be equivalent in the end.

Reply:
addressed